src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -298,8 +300,17 @@ def visit_Call(self, node: foast.Call, **kwargs) -> foast.Call:
                     new_dims.extend(target_dims)
             new_type = ct.FieldType(dims=new_dims, dtype=new_func.type.dtype)
             return foast.Call(func=new_func, args=new_args, location=node.location, type=new_type)
-        return foast.Call(
-            func=new_func, args=self.visit(node.args, **kwargs), location=node.location
+        elif isinstance(new_func.type, ct.FunctionType):
+            return foast.Call(
+                func=new_func,
+                args=self.visit(node.args, **kwargs),
+                type=new_func.type.returns,
+                location=node.location,
+            )
```

Comment:
The type deduction part might be out of sync with the recently merged PR adding reductions and builtin functions. Please, take a look before merging: https://github.com/GridTools/gt4py/pull/640/files#diff-5a2d8f3beae7064a7845cfddc917538e55d53d9d3694a978d2e8b2c5a72ab3ae

Reply:
I checked and this still works. Nonetheless the type deduction of FOAST is incomplete in the sense that not all types are deduced (for sure after the reductions PR). We should fix this at some point, but this requires either a function to compute the resulting type of a builtin (not so hard) or a generic algorithm and type parameters (nice, but more work).